lib_test/test_connect.ml

```diff
@@ -83,11 +83,40 @@ module Test_connect (B : Vnetif_backends.Backend) = struct
 
     Lwt.return_unit
 
+  let test_tcp_connect_to_closed_port () =
+    or_error "console" Console.connect "console" >>= fun c ->
+    let timeout = 15.0 in
+    Lwt.pick [
+      (Lwt_unix.sleep timeout >>= fun () ->
+       fail "connect test timed out after %f seconds" timeout) ;
+
+      (V.create_stack c backend server_ip netmask [gw] >>= fun s1 ->
+       (* register no listeners *)
+       V.Stackv4.listen s1);
+
+      (Lwt_unix.sleep 0.1 >>= fun () ->
```

Comment:
I dislike unit tests with real sleeps in them. Apart from being unreliable, when building lots of packages they all add up...
But still better than not having a test, so not a blocker for the PR.